### PR TITLE
Amazon Location: use fetchMap in Getting Started page

### PR DESCRIPTION
**Pull request by felixpalmer**
In order to leverage the shared cache, non-deck.gl-based clients need to specify the `v=3.1` parameter and use only  header based authentication. Related: https://github.com/visgl/deck.gl/pull/7287/

### CODE REVIEW

**Review by alasarr — CHANGES_REQUESTED** [review]
Just realized we should change this way in mapbox gl and amazon location.

We developed these guides when we only had tilesets support and it's today only compatible when the backend returns data in MVT, so: 

- Tables will fail in all providers, but PostgreSQL
- Spatial indexes in all providers
- Tables will fail in all providers, but BigQuery

I propose to force always to use deck.gl, so we need to change these examples and do the same as we do with Google Maps.

**Comment by felixpalmer** [comment]
@alasarr OK, I will rework this PR to always use deck. I think it ultimately makes it simpler, as we are then promoting one correct way of working with our backend

**Comment by felixpalmer** [comment]
@alasarr I've updated the Amazon Location docs so that:

- fetchMap is used in the getting started page
- The Hello World example also uses fetchMap (same example as above)
- A new CartoLayer example is included also

I think this approach simplifies the docs and shows how users can get started with CARTO with only a few lines of code.

If you're happy with this approach, I can do the same for the Mapbox

File: app/content/amazon-location/examples/cartolayer.html [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        container: "map",
+        center: [20, 49], 
+        zoom: 4, 
+        style: mapName,
+        transformRequest,
+      });
+
+      map.addControl(new maplibregl.NavigationControl(), "top-left");
+
+      deck.carto.setDefaultCredentials({accessToken: 'eyJhbGciOiJIUzI1NiJ9.eyJhIjoiYWNfbHFlM3p3Z3UiLCJqdGkiOiI1YjI0OWE2ZCJ9.Y7zB30NJFzq5fPv8W5nkoH5lPXFWQP0uywDtqUg8y8c'});
```
Comment (alasarr) [inline app/content/amazon-location/examples/cartolayer.html:76]: We could add the token in a variable at the top to make it easier

File: app/content/amazon-location/examples/getting-started/step-2.html [hunk 2]
```diff
@@ -61,59 +65,18 @@
...
-                [{"zoom": 0, "value": 322}, "rgba(176, 79, 128, 1)"],
-                [{"zoom": 0, "value": 341}, "rgba(215, 39, 128, 1)"],
-                [{"zoom": 0, "value": 360}, "rgba(255, 0, 128, 1)"]
-              ]
-            }
-          }
-        }
-      );
+      const cartoMapId = 'ff6ac53f-741a-49fb-b615-d040bc5a96b8';
+      const {layers} = await deck.carto.fetchMap({cartoMapId});
```
Comment (alasarr) [inline app/content/amazon-location/examples/getting-started/step-2.html:77]: 👍

**Review by alasarr — APPROVED** [review]
+1 to the idea of using fetchMap as default and adding the CartoLayer section to the doc.

Thanks a lot for the proposal, it looks great

**Pull request merged**